### PR TITLE
Clean up CouchConfig class

### PR DESCRIPTION
**Pull request by rhyshort**
Clean up CouchConfig class by removing options that were not honoured.

Split out from PR #181 

reviewer @mikerhodes 
reviewer @tomblench

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 - [BREAKING CHANGE] Moved code for encryption on Android to the
   `sync-android-encryption` subproject. See the instructions in the [README](https://github.com/cloudant/sync-android/blob/master/README.md)
   file for how to include `sync-android-encryption` in your project.
+- [BREAKING CHANGE] Removed Configuration options from CouchConfig which are not used
```
Comment (mikerhodes) [inline CHANGES.md]: Can you link to the commit here so people can see what changed?

**Comment by mikerhodes** [comment]
Looks generally good -- though how difficult would it be to instead make the options used? Only worth doing, ofc, if the options were public in the first place.

**Comment by tomblench** [comment]
`HTTPUrlConnection` doesn't have as many low level settings, but it's possible to adjust timeouts etc  through that class, and a few other low-level settings through properties: http://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html#MiscHTTP

Perhaps these could be addressed in a separate ticket?

File: CHANGES.md (hunk 1 above)
Comment (rhyshort) [inline CHANGES.md]: done in commit cfd7730

**Comment by rhyshort** [comment]
I'd side on handling it in a separate ticket, that way we can look at which options we can actually provide, for example the connection per host can't be provided until we deal with the keep alive issue.

**Comment by mikerhodes** [comment]
Sounds good -- let's add some of the options we can do (I guess things like number of retries for network failures might go here).

+1 otherwise.

File: CHANGES.md (hunk 1 above)
Comment (tomblench) [inline CHANGES.md]: Configuration and CouchConfig should be in backticks to make it clearer.

'see commit' should either be in its own sentence or in brackets

Comment (tomblench) [inline CHANGES.md]: also 'see commit xxx for details' is clearer

**Comment by tomblench** [comment]
+1 after my minor comments addressed

**Pull request merged**